feat: Top collections - landing page v3

hassnian opened the pull request:
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring


## Context

- [x] Closes #9309

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)


## Screenshot 📸

- [x] My fix has changed **something** on UI; 

![CleanShot 2024-02-11 at 10 38 51@2x](https://github.com/kodadot/nft-gallery/assets/44554284/7eedd70a-2a32-45ad-8a29-ad092fa087c1)

hassnian commented:
no fancy layout

on telegram @exezbcz `lets make it generic layout`

hassnian commented:
also can we change the default time range to 30 days ?

I think makes more sense otherwise the floor price diff is 0 

![CleanShot 2024-02-10 at 16 19 43@2x](https://github.com/kodadot/nft-gallery/assets/44554284/ca06474a-6744-4393-9037-af80cbb1948d)

![CleanShot 2024-02-10 at 16 21 47@2x](https://github.com/kodadot/nft-gallery/assets/44554284/838c24ce-998a-4cc2-b1af-cd68b7da5d71)

exezbcz commented:
@hassnian sure

yangwao commented:
can be % moved on the left? so it's consistent and not jumping
![image](https://github.com/kodadot/nft-gallery/assets/5887929/4c3fba73-c167-46a5-ba51-d0a5a0f53b62)
![image](https://github.com/kodadot/nft-gallery/assets/5887929/2e3f207c-55b2-488f-9aed-3092dae43bd0)

yangwao commented:
and I would remove that one 3 rows where 3rd row has only 2 items, it's bit ugly

https://github.com/kodadot/nft-gallery/assets/5887929/82a6c80d-84c3-44aa-8a1b-88f5b351105f

exezbcz commented:
> and I would remove that one 3 rows where 3rd row has only 2 items, it's bit ugly

it could be 3 rows of 4 items

hassnian commented:
> can be % moved on the left? so it's consistent and not jumping ![image](https://private-user-images.githubusercontent.com/5887929/303913233-4c3fba73-c167-46a5-ba51-d0a5a0f53b62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3MTI5NTUsIm5iZiI6MTcwNzcxMjY1NSwicGF0aCI6Ii81ODg3OTI5LzMwMzkxMzIzMy00YzNmYmE3My1jMTY3LTQ2YTUtYmE1MS1kMGE1YTBmNTNiNjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTJUMDQzNzM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmUxYjVhYmY3ZGQ3ZjliNjExMjdhNGUyYjRmMzI3NDhmY2Y2ZDA1OGU4ZjZjYjg4NWZmMTYxZGYyNGQ1ZDYwYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6qEM1axUA6-1_5uOg4vCdOoH_JND3DgrNhsOcSfSblc) ![image](https://private-user-images.githubusercontent.com/5887929/303913236-2e3f207c-55b2-488f-9aed-3092dae43bd0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3MTI5NTUsIm5iZiI6MTcwNzcxMjY1NSwicGF0aCI6Ii81ODg3OTI5LzMwMzkxMzIzNi0yZTNmMjA3Yy01NWIyLTQ4OGYtOWFlZC0zMDkyZGFlNDNiZDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTJUMDQzNzM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQwMGQ0MWE5MDczMGZiZDY0NzgzNjI1ZDVjMDM3Nzg0MDdkZWE0YmQzNDk1ZTBiOGVjZWM0OTk4YjdlNzA0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3GbGXY3mAi-6-2xk7Krq8eN5d6tbliZyO05US-JvAgo)

fixed


> > and I would remove that one 3 rows where 3rd row has only 2 items, it's bit ugly
> 
> it could be 3 rows of 4 items

fixed

exezbcz commented:
lets add bigger margin and then its perfect 


![image](https://github.com/kodadot/nft-gallery/assets/90852205/1dbdaa78-217b-4156-82ce-7ba85447b9cb)

hassnian commented:
> lets add bigger margin and then its perfect
> 

fixed, check again,  thanks

hassnian commented:
## Side note 

this pr does change the number of cols that are created on other grids like nft grid and collection ,grid (check between canary and this pr you'll see the difference)  

prev it didn't take the column gaps into consideration when calculating the amount of cols

now it does,   chaging the column amount **only when the min items size is not met anymore** 

@exezbcz just fyi 
@prury don't know if this affects any of your tests

roiLeo commented on components/landing/topCollections/TopCollectionsCard/TopCollectionsCard.scss:
```diff
@@ -0,0 +1,38 @@
+@import '@/assets/styles/abstracts/variables';
```
why need variables here?

hassnian replied:
removed

roiLeo commented on components/landing/topCollections/TopCollectionsCard/TopCollectionsCard.scss:
```diff
@@ -0,0 +1,38 @@
+@import '@/assets/styles/abstracts/variables';
+
+.top-collection-card , .top-collection-card-skeleton {
```
can't we replace this file with tailwind classes?

```suggestion
.top-collection-card, 
.top-collection-card-skeleton {
```

hassnian replied:
fixed

roiLeo commented on components/landing/topCollections/TopCollections.vue:
```diff
@@ -1,8 +1,10 @@
-    <div class="flex justify-between mobile">
-      <div class="title is-2">{{ $t('general.topCollectionsHeading') }}</div>
-      <div class="top-collection-controls buttons items-start pt-2">
+    <div class="flex justify-between mobile !pb-9 md:!pb-10">
+      <div class="title is-2 md:!mb-0">
+        {{ $t('general.topCollectionsHeading') }}
+      </div>
+      <div class="top-collection-controls buttons items-start pt-2 !mb-0">
```
why `!important`?

hassnian replied:
adding any padding/margin class ends up using the bulma class and not the tailwindcss one , not needed here since bulma only goes up to -6 , I was changing the amount and forgot remove it

preschian commented on assets/styles/components/_top-collections.scss:
would be good if we can delete this file

preschian commented on components/landing/topCollections/TopCollectionsCard/TopCollectionsCard.scss:
```diff
@@ -0,0 +1,37 @@
+.top-collection-card,
```
Create a shared component instead of shared scss files?

hassnian replied:
done pls check

preschian replied:
prefer something like this, then we can inline tailwind classes

```vue
<template>
  <div
    class="top-collection-card w-full border border-card-border-color hover:border-border-color bg-background-color group">
    <nuxt-link
      class="flex flex-col hover:text-text-color"
      rel="nofollow noopener noreferrer"
      :to="`${collection?.id && `/${urlPrefix}/collection/${collection.id}`}`">
      <div
        class="top-collection-card-banner group-hover:opacity-[0.85] bg-center bg-cover"
        :style="{ backgroundImage: `url(${collection?.image})` }"
        :class="`${!collection && 'bg-k-shade animate-pulse'}`">
        <div class="top-collection-card-banner-avatar-container">
          <BasicImage
            custom-class="top-collection-card-banner-avatar-inner [&>img]:border [&>img]:border-card-border-color !p-0"
            :src="collection?.image || placeholder" />
        </div>
      </div>

      <div class="top-collection-card-info border-t border-card-border-color">
        <p
          class="collection-name font-bold text-base whitespace-nowrap text-center truncate">
          <span v-if="collection?.name">{{ collection.name }}</span>
          <span v-else class="inline-block">
            <NeoSkeleton no-margin :rounded="false" width="160" height="24" />
          </span>
        </p>

        <div class="info-fields">
          <div>
            <p class="capitalize text-k-grey text-xs">
              <span v-if="collection?.floorPrice || collection?.floor">
                {{ $t('price') }}
              </span>
              <NeoSkeleton
                v-else
                no-margin
                :rounded="false"
                width="40"
                height="12" />
            </p>

            <div
              v-if="collection?.floorPrice || collection?.floor"
              class="flex gap-2 items-center max-md:flex-row-reverse">
              <CommonTokenMoney
                :value="collection.floorPrice || collection.floor"
                inline
                :round="2" />

              <div v-if="formattedDiffPercent" :class="color" class="text-xs">
                {{ formattedDiffPercent }}
              </div>
            </div>
            <NeoSkeleton
              v-else
              no-margin
              :rounded="false"
              width="80"
              height="20"
              class="mt-1" />
          </div>

          <div>
            <p class="capitalize text-k-grey text-end text-xs">
              <span v-if="collection">{{ $t('volume') }}</span>
              <NeoSkeleton
                v-else
                no-margin
                :rounded="false"
                width="40"
                height="12" />
            </p>
            <div class="flex gap-2 items-center">
              <CommonTokenMoney
                v-if="collection"
                :value="volume"
                inline
                :round="1" />
              <NeoSkeleton
                v-else
                no-margin
                :rounded="false"
                width="80"
                height="20" />
            </div>
          </div>
        </div>
      </div>
    </nuxt-link>
  </div>
</template>
```

preschian replied:
that one is `TopCollectionsCard` component

hassnian replied:
fixed

prury commented:
> 
>  @prury don't know if this affects any of your tests

it does not, ty

> ## Side note
> 
> this pr does change the number of cols that are created on other grids like , nft grid and collection grid (check between canary and this pr you'll see the difference)
> 

true, it changes, and there's something strange as well, look at the number of results: 

https://github.com/kodadot/nft-gallery/assets/36627808/0f0b63dc-297f-4472-a475-64bb4ccbf2c0

exezbcz commented:
deploy: 

![image](https://github.com/kodadot/nft-gallery/assets/90852205/db3f27e0-aa45-4191-9b61-49ae85e9a0fa)



canary: 

![image](https://github.com/kodadot/nft-gallery/assets/90852205/e1be0889-de17-446e-b8a5-694f04652451)


> prev it didn't take the column gaps into consideration when calculating the amount of cols

if we will get more precise columns and cards that don't drop below the minimum width, I am for it!

preschian commented on components/landing/topCollections/TopCollectionsGrid.vue:
```diff
@@ -0,0 +1,42 @@
+<template>
+  <DynamicGrid
```
Is it possible to use css instead? demo here https://play.tailwindcss.com/HY9KzaQxiJ?layout=preview&size=1750x720

hassnian replied:
but this does not take the item min width into consideration.

preschian replied:
enlighten me more pls, looks same to me

hassnian replied:
### min width of these cards is 256px

in the demo that you provided before going to 3 cols the size of each item is 224px

![CleanShot 2024-02-13 at 09 08 30@2x](https://github.com/kodadot/nft-gallery/assets/44554284/075a4bed-5587-4e46-9db5-957a646c486b)

using dynamic grid , breaks only when 256px min width is not met anymore 

![CleanShot 2024-02-13 at 09 10 36@2x](https://github.com/kodadot/nft-gallery/assets/44554284/7cceedad-02ac-4a86-9b7e-afb3dfc01b2a)

preschian replied:
try to put this code snippet below your `v-for="(collection, index) in collections"`. and compare with your card, looks same to me

```vue
<div v-for="n in 12" :key="n" class="h-40 w-full bg-gray-800 text-white">
  {{ n }}
</div>
```

preschian replied:
oopps, wait

preschian replied:
aahh yes, the items are not even on the row with css

exezbcz commented:
green check for me! good job

roiLeo commented on components/landing/topCollections/TopCollections.vue:
```diff
@@ -1,8 +1,10 @@
-      <div class="title is-2">{{ $t('general.topCollectionsHeading') }}</div>
-      <div class="top-collection-controls buttons items-start pt-2">
+    <div class="flex justify-between mobile pb-9 md:pb-10">
+      <div class="title is-2 md:!mb-0">
```
`title is-2` will be remove (#9013) please could you use tailwind helper instead?

hassnian replied:
done

roiLeo commented on components/landing/topCollections/TopCollections.vue:
```diff
@@ -1,8 +1,10 @@
+      <div class="title is-2 md:!mb-0">
+        {{ $t('general.topCollectionsHeading') }}
+      </div>
+      <div class="top-collection-controls buttons items-start pt-2 !mb-0">
```
`buttons` is a Bulma class I would use tailwind equivalent 

```css
.buttons {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
}
```

hassnian replied:
done

roiLeo reviewed (approved):
No a fan of DynamicGrid js, should be doable with only css but can be in a follow up.

otherwise lgtm

hassnian commented:
> No a fan of DynamicGrid js, should be doable with only css

related? https://github.com/kodadot/nft-gallery/pull/9336#discussion_r1487101461

> but can be in a follow up.

sure , let's do that

yangwao commented:
Thanks!
pay 80 usd

yangwao commented:
😍 Perfect, I’ve sent the payout 
💵 $80 @ 7.65 USD/DOT ~ 10.458 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x150d44dee3ded30d38ea494716441347bd8070b007d9e79ddcf03ee0d9a3d6d1](https://polkadot.subscan.io/extrinsic/0x150d44dee3ded30d38ea494716441347bd8070b007d9e79ddcf03ee0d9a3d6d1)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
